backend/domain_types/src/types.rs

```diff
@@ -1669,6 +1706,18 @@ impl ForeignTryFrom<(PaymentServiceAuthorizeRequest, Connectors, &MaskedMetadata
                 ))
             })?;
 
+        let order_details = if value.order_details.is_empty() {
+            None
+        } else {
+            Some(
+                value
+                    .order_details
+                    .into_iter()
+                    .map(OrderDetailsWithAmount::foreign_try_from)
+                    .collect::<Result<Vec<_>, _>>()?,
+            )
+        };
```

Comment:
```suggestion
let order_details = (!value.order_details.is_empty()).then(|| {
    value
        .order_details
        .into_iter()
        .map(OrderDetailsWithAmount::foreign_try_from)
        .collect::<Result<Vec<_>, _>>()
}).transpose()?;
```
Maybe we can try writing it this way??